### PR TITLE
Generate token json

### PR DESCRIPTION
**Pull request by NathanSadler**
## Why?

We wanted to make it possible to easily transfer changed styles to Figma

## What Changed

What changed in this PR?

* [x] Create script that exports tokens as a json file

## Sanity Check

* [x] Have you updated any usage of changed tokens?
* [x] Have you updated the `docs/token_structure.json` file?
* [x] Have you updated the docs with any component changes?
* [] Do you need to update the package version?

### CODE REVIEW

File: generate_token_json.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+const fs = require('fs')
+const { join } = require('path')
```
Comment (NathanSadler) [inline generate_token_json.js:2]: remove this

File: generate_token_json.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  return keys
+}
+
+function addToJson(keys, data, value) {
+  let currentKey, remainingKeys
+
+  if (isHyphenated(keys, 2)) {
+    [currentKey, remainingKeys] = [keys.slice(0, 2).join('-'), keys.slice(2)]
+  } else if (isHyphenated(keys, 3)) {
+    [currentKey, remainingKeys] = [keys.join('-'), keys.slice(3)]
```
Comment (NathanSadler) [inline generate_token_json.js:33]: should a .slice(0, 3) get added here?

File: docs/token_structure.json [hunk 3]
```diff
@@ -43,23 +43,21 @@
         "s": "",
```
Comment (Jeremy-Walton) [inline docs/token_structure.json]: This file can go away

File: generate_token_json.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  'width-large', 'width-x-large', 'card-outline',
+  'x-small', '2x-small', '3x-small', '4x-small', '5x-small',
+  'x-large', '2x-large', '3x-large', '4x-large', '5x-large',
+  'extra-light', 'extra-bold', 'semi-bold',
+  'dialog-backdrop', 'dialog-content', 'alert-group', 'on-minus-max', 'on-minus-1', 'on-minus-2', 'on-minus-3',
+  'on-plus-min', 'on-plus-max', 'on-plus-1', 'on-plus-2', 'on-plus-3', 'on-hover', 'on-base', 'letter-spacing', 'scale-unit',
+  'plus-1', 'plus-2', 'plus-3', 'plus-max', 'minus-1', 'minus-2', 'minus-3', 'minus-max', 'input-height'
+]
+
+const LUMINOSITY_KEYS = [...Array(101).keys()].map((luminosity) => `luminosity-${luminosity}`)
```
Comment (Jeremy-Walton) [inline generate_token_json.js:19]: What does the Array(101) do? It's not obvious from this code what that means

File: package.json [hunk 5]
```diff
@@ -31,6 +31,7 @@
     "@RoleModel:registry": "https://npm.pkg.github.com"
   },
   "dependencies": {
+    "@rolemodel/rolemodel-design-system": "^0.0.6-beta",
```
Comment (Jeremy-Walton) [inline package.json:34]: Why are we adding ourselves as a dependency?

File: package-lock.json [hunk 6]
```diff
@@ -9,6 +9,7 @@
       "version": "0.0.6-beta",
       "license": "MIT",
       "dependencies": {
+        "@rolemodel/rolemodel-design-system": "^0.0.6-beta",
```
Comment (Jeremy-Walton) [inline package-lock.json:12]: This isn't needed

File: package.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
     "build": "npm run build:css && npm run build:scss && npm run build:tokens",
     "build:css": "sass src/rolemodel-design-system.scss dist/css/rolemodel-design-system.css --load-path=node_modules/",
     "build:scss": "cp -r src dist/scss",
-    "build:tokens": "mkdir -p dist/tokens && cp docs/token_structure.json dist/tokens/tokens.json"
+    "build:tokens": "node generate_token_json"
```
Comment (Jeremy-Walton) [inline package.json:11]: I would call this script `build_token_json` to be consistent with the command name

File: generate_token_json.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
+const fs = require('fs')
```
Comment (Jeremy-Walton) [inline generate_token_json.js:1]: I would rename this file to `build_token_json.js`

File: generate_token_json.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    default:
+      data[currentKey] = addToJson(remainingKeys, data[currentKey], value)
+      break
+  }
+
+  return data
+}
+
+try {
+  const tokenDirectory = './src/core/tokens'
```
Comment (Jeremy-Walton) [inline generate_token_json.js:55]: Is there a way to make these script arguments so they can be passed in when you call the script? That way someone could change them easily if needed?

**Review by Jeremy-Walton — CHANGES_REQUESTED** [review]

File: generate_token_json.js (hunk 4 above)
Comment (NathanSadler) [reply generate_token_json.js]: I added a comment to explain things

File: yarn.lock [hunk 10]
```diff
@@ -0,0 +1,149 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
[the remark this answers is not in the document]
Comment (NathanSadler) [reply yarn.lock]: Done.

File: package.json (hunk 5 above)
Comment (NathanSadler) [reply package.json]: Okay, removed.

File: package-lock.json (hunk 6 above)
Comment (NathanSadler) [reply package-lock.json]: fixed

File: package.json (hunk 7 above)
Comment (NathanSadler) [reply package.json]: ok, renamed.

File: generate_token_json.js (hunk 8 above)
Comment (NathanSadler) [reply generate_token_json.js]: done

File: generate_token_json.js (hunk 9 above)
Comment (NathanSadler) [reply generate_token_json.js]: Yes, and it has been implemented

File: generate_token_json.js (hunk 2 above)
Comment (NathanSadler) [reply generate_token_json.js]: It doesn't need it.

**Review by Jeremy-Walton — APPROVED** [review]

**Pull request merged**